src/ServicePulse.Host/vue/src/components/failedmessages/MessageView.vue

```diff
@@ -302,6 +306,35 @@ function exportMessage() {
   useShowToast("info", "Info", "Message export completed.");
 }
 
+function showEditAndRetryModal() {
+  showEditRetryModal.value = true;
+  return stopRefreshInterval();
+}
+
+function cancelEditAndRetry() {
+  showEditRetryModal.value = false;
+  loadFailedMessage(); //Reset the message object when canceling the edit & retry modal
+  return startRefreshInterval();
+}
+
+function confirmEditAndRetry() {
+  showEditRetryModal.value = false;
+  useShowToast("info", "Info", `Retrying the edited message ${id} ...`);
+  return startRefreshInterval();
+}
+
+function startRefreshInterval() {
+  refreshInterval = setInterval(() => {
+    loadFailedMessage();
+  }, 5000);
+}
```

Comment:
Should we clear the interval if it exists just in case to prevent accidental memory leaks?

```suggestion
function startRefreshInterval() {
  stopRefreshInterval();

  refreshInterval = setInterval(() => {
    loadFailedMessage();
  }, 5000);
}
```